torch/_dynamo/metrics_context.py

```diff
@@ -126,3 +126,42 @@ def add_to_set(self, metric: str, value: Any) -> None:
         if metric not in self._metrics:
             self._metrics[metric] = set()
         self._metrics[metric].add(value)
+
+
+class RuntimeMetricsContext:
+    def __init__(self, on_exit: OnExitType):
```

Comment:
I'm not a big fan of this indirection haha, mostly cause now I have to dig to find where on_exit is defined

Reply:
Established in `MetricsContext`, the thinking being that MetricsContext` is (currently) generalized and completely agnostic of how it's used to ultimately log. Which also made the unit tests actual unit tests and easy to write without any mocking. Happy to change both if it bothers you.